diff --git a/esp-idf/components/driver/spi_slave.c b/esp-idf/components/driver/spi_slave.c
index 82fc99c10..a479af469 100644
--- a/esp-idf/components/driver/spi_slave.c
+++ b/esp-idf/components/driver/spi_slave.c
@@ -206,38 +206,24 @@ esp_err_t spi_slave_initialize(spi_host_device_t host, const spi_bus_config_t *b
     spihost[host]->hw->ctrl.wr_bit_order = (slave_config->flags & SPI_SLAVE_TXBIT_LSBFIRST) ? 1 : 0;
 
     const int mode = slave_config->mode;
+    bool nodelay = false;
     if (mode == 0) {
-        //The timing needs to be fixed to meet the requirements of DMA
-        spihost[host]->hw->pin.ck_idle_edge = 1;
-        spihost[host]->hw->user.ck_i_edge = 0;
-        spihost[host]->hw->ctrl2.miso_delay_mode = 0;
-        spihost[host]->hw->ctrl2.miso_delay_num = 0;
-        spihost[host]->hw->ctrl2.mosi_delay_mode = 2;
-        spihost[host]->hw->ctrl2.mosi_delay_num = 2;
-    } else if (mode == 1) {
-        spihost[host]->hw->pin.ck_idle_edge = 1;
-        spihost[host]->hw->user.ck_i_edge = 1;
-        spihost[host]->hw->ctrl2.miso_delay_mode = 2;
-        spihost[host]->hw->ctrl2.miso_delay_num = 0;
-        spihost[host]->hw->ctrl2.mosi_delay_mode = 0;
-        spihost[host]->hw->ctrl2.mosi_delay_num = 0;
-    } else if (mode == 2) {
-        //The timing needs to be fixed to meet the requirements of DMA
         spihost[host]->hw->pin.ck_idle_edge = 0;
         spihost[host]->hw->user.ck_i_edge = 1;
-        spihost[host]->hw->ctrl2.miso_delay_mode = 0;
-        spihost[host]->hw->ctrl2.miso_delay_num = 0;
-        spihost[host]->hw->ctrl2.mosi_delay_mode = 1;
-        spihost[host]->hw->ctrl2.mosi_delay_num = 2;
-    } else if (mode == 3) {
+        spihost[host]->hw->ctrl2.miso_delay_mode = nodelay ? 0 : 2;
+    } else if (mode == 1) {
         spihost[host]->hw->pin.ck_idle_edge = 0;
         spihost[host]->hw->user.ck_i_edge = 0;
-        spihost[host]->hw->ctrl2.miso_delay_mode = 1;
-        spihost[host]->hw->ctrl2.miso_delay_num = 0;
-        spihost[host]->hw->ctrl2.mosi_delay_mode = 0;
-        spihost[host]->hw->ctrl2.mosi_delay_num = 0;
+        spihost[host]->hw->ctrl2.miso_delay_mode = nodelay ? 0 : 1;
+    } else if (mode == 2) {
+        spihost[host]->hw->pin.ck_idle_edge = 1;
+        spihost[host]->hw->user.ck_i_edge = 0;
+        spihost[host]->hw->ctrl2.miso_delay_mode = nodelay ? 0 : 1;
+    } else if (mode == 3) {
+        spihost[host]->hw->pin.ck_idle_edge = 1;
+        spihost[host]->hw->user.ck_i_edge = 1;
+        spihost[host]->hw->ctrl2.miso_delay_mode = nodelay ? 0 : 2;
     }
-
     /* Silicon issues exists in mode 0 and 2 with DMA, change clock phase to
      * avoid dma issue. This will cause slave output to appear at most half a
      * spi clock before
diff --git a/esp-idf/components/soc/esp32/gpio_periph.c b/esp-idf/components/soc/esp32/gpio_periph.c
index 3534b7b3e..1e8e5557a 100644
--- a/esp-idf/components/soc/esp32/gpio_periph.c
+++ b/esp-idf/components/soc/esp32/gpio_periph.c
@@ -35,7 +35,7 @@ const uint32_t GPIO_PIN_MUX_REG[GPIO_PIN_COUNT] = {
     IO_MUX_GPIO17_REG,
     IO_MUX_GPIO18_REG,
     IO_MUX_GPIO19_REG,
-    0,
+    IO_MUX_GPIO20_REG,
     IO_MUX_GPIO21_REG,
     IO_MUX_GPIO22_REG,
     IO_MUX_GPIO23_REG,
